test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/CpsPackageReferenceProjectTests.cs

```diff
@@ -4053,9 +4053,36 @@ public async Task GetInstalledAndTransitivePackagesAsync_ProjectReferenceWithSin
 
             // Assert
             Assert.Empty(result.InstalledPackages); // No installed packages
-            // Foreign transitive packages in assets file don't have dependencies.
-            // Those are not considered transitive packages
-            Assert.Empty(result.TransitivePackages);
+            Assert.Equal(1, result.TransitivePackages.Count);
+        }
+
+        [Fact]
+        public async Task GetInstalledAndTransitivePackagesAsync_MutipleProjectReferencesWithSinglePackage_EmptyInstalledAndTransitivePackagesAsync()
+        {
+            // Project3 -> Project2 ->  PackageA (1.0.0)
+            //                          Project1 -> PackageB (1.0.0)
+            // PackageA (1.0.0) -> PackageB (1.0.0)
```

Comment:
Doesn't this test the scenario with both a project and package referencing Package B?

In general, for transitive packages, we have 3 options: 
* Through package only
* Through project only
* Through project & package.